internal/trie/node/branch_encode.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package node
+
+import (
+	"bytes"
+	"fmt"
+	"hash"
+	"io"
+
+	"github.com/ChainSafe/gossamer/internal/trie/codec"
+	"github.com/ChainSafe/gossamer/internal/trie/pools"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// ScaleEncodeHash hashes the node (blake2b sum on encoded value)
+// and then SCALE encodes it. This is used to encode children
+// nodes of branches.
+func (b *Branch) ScaleEncodeHash() (encoding []byte, err error) {
+	buffer := pools.DigestBuffers.Get().(*bytes.Buffer)
+	buffer.Reset()
+	defer pools.DigestBuffers.Put(buffer)
+
+	err = b.hash(buffer)
+	if err != nil {
+		return nil, fmt.Errorf("cannot hash branch: %w", err)
+	}
+
+	encoding, err = scale.Marshal(buffer.Bytes())
+	if err != nil {
+		return nil, fmt.Errorf("cannot scale encode hashed branch: %w", err)
+	}
+
+	return encoding, nil
+}
+
+func (b *Branch) hash(digestBuffer io.Writer) (err error) {
+	encodingBuffer := pools.EncodingBuffers.Get().(*bytes.Buffer)
+	encodingBuffer.Reset()
+	defer pools.EncodingBuffers.Put(encodingBuffer)
+
+	err = b.Encode(encodingBuffer)
+	if err != nil {
+		return fmt.Errorf("cannot encode leaf: %w", err)
+	}
+
+	// if length of encoded leaf is less than 32 bytes, do not hash
```

Comment:
```suggestion
	// if length of encoded node is less than 32 bytes, do not hash
```

Reply:
Change to `branch` in https://github.com/ChainSafe/gossamer/pull/2077/commits/865aae6d143ec3137c92d30a7ce7235b96bb41bb